src/python/grpcio/grpc/_cython/_cygrpc/call.pyx.pxi

```diff
@@ -106,6 +107,8 @@ cdef class Call:
   def __dealloc__(self):
     if self.c_call != NULL:
       grpc_call_destroy(self.c_call)
+    grpc_shutdown()
```

Comment:
If each destructor was getting called prior to program exit, I wouldn't expect numerous memory leaks without the init/shutdown parity.  The only reason I can come up with as to why the memory leak messages are gone after this change is that ```g_initializations``` never reaches 0.

I suspect that something with PyPy is either not calling the __dealloc__ method or not running garbage collection on exit.  We can check this theory with gdb or GRPC_TRACE='api'